### PR TITLE
Enable `no-transport` mode as startup parameter

### PR DESCRIPTION
**Pull request by PatStLouis**
This PR adds a `--no-transport` startup parameter for making the agent start without requiring the `--enpoint`, `--inbound-transport` and `--outbound-transport` to be set.

The idea was shared during an aca-pug meeting and there was no objection. This can greatly simplify deployment for controllers that do no require didcomm messaging.

Currently it just overrides what is required for startup.

@dbluhm @swcurran @ianco

### CODE REVIEW

**Comment by dbluhm** [comment]
Like I mentioned in the ACA-PUG call, I like where this is going. Having the ability to use ACA-Py's credential, DID Resolution, and other capabilities without necessarily using the DIDComm capabilities is really interesting.

I think you're already headed in this direction but if no transport is enabled, it is likely not necessary to register any of the protocols.

File: aries_cloudagent/core/conductor.py [hunk 1]
```diff
@@ -191,20 +191,22 @@ async def setup(self):
...
+            context.injector.bind_instance(
+                InboundTransportManager, self.inbound_transport_manager
+            )
 
-        # Register all outbound transports
-        self.outbound_transport_manager = OutboundTransportManager(
-            self.root_profile, self.handle_not_delivered
-        )
-        await self.outbound_transport_manager.setup()
+        if not context.settings.get("transport.no_transport"):
```
[the remark this answers is not in the document]
Comment (PatStLouis) [reply aries_cloudagent/core/conductor.py]: Changed it, good catch

**Review by dbluhm — DISMISSED** [review]

**Comment by dbluhm** [comment]
Integration tests are consistently failing. The question is whether that's a result of the changes in this PR or not.

```
 Failing scenarios:
  features/0586-sign-transaction.feature:124  endorse a schema and cred def transaction, write to the ledger, issue and revoke a credential, manually invoking each endorsement endpoint -- @4.1 
  features/0586-sign-transaction.feature:234  endorse a schema and cred def transaction, write to the ledger, issue and revoke a credential, with auto endorsing workflow -- @1.1 
  features/0586-sign-transaction.feature:245  endorse a schema and cred def transaction, write to the ledger, issue and revoke a credential, with auto endorsing workflow -- @3.1 
```

It doesn't seem like these changes should impact these scenarios... @jamshale any insights?

**Comment by jamshale** [comment]
I noticed this but have not dug into it. I'm not sure how this is effecting endorsement but it must be as integration tests are passing on the nightly runs and for other pr's. I can try and see if I can figure anything out.

**Comment by PatStLouis** [comment]
I will have a look, no additional tests were added and no changes are expected if the flag is not added. Maybe there's a bad condition check somewhere.

**Comment by jamshale** [comment]
I haven't tried to debug it yet but these failing tests are trying to use the `/ledger/register-nym` endpoint via the `register_did` method in `demo/runners/support/agent.py` and getting a 400 error.

**Comment by PatStLouis** [comment]
@dbluhm @jamshale I think I found the error, copy paste mistake. I was registering the ledger as a package and not a plugin. Lets see how the test go now

**Comment by PatStLouis** [comment]
Tests passed, updating the branch

**Review by jamshale — APPROVED** [review]

File: aries_cloudagent/config/default_context.py [hunk 2]
```diff
@@ -116,11 +116,15 @@ async def load_plugins(self, context: InjectionContext):
...
-        plugin_registry.register_package("aries_cloudagent.protocols")
+        if not self.settings.get("transport.disabled"):
+            plugin_registry.register_package("aries_cloudagent.protocols")
 
         # Currently providing admin routes only
         plugin_registry.register_plugin("aries_cloudagent.holder")
-        plugin_registry.register_plugin("aries_cloudagent.ledger")
+
+        if not self.settings.get("ledger.disabled"):
+            plugin_registry.register_plugin("aries_cloudagent.ledger")
```
[the remark this answers is not in the document]
Comment (PatStLouis) [reply aries_cloudagent/config/default_context.py:126]: I can revert this, it's not crucial to what I was trying to achieve here. I'll submit a PR, thanks for flagging this!

[the remark this answers is not in the document]
Comment (ff137) [reply aries_cloudagent/config/default_context.py:126]: Cool! Picked it up when generating openapi client off nightly build. Thanks! 👍

**Pull request merged**